collector/config/config.go

```diff
@@ -26,3 +39,76 @@ type Config struct {
 	MaxGRPCRetries         uint32        `mapstructure:"max_grpc_retries"`
 	MaxRPCMsgSize          int           `mapstructure:"max_rpc_msg_size"`
 }
+
+var _ xconfmap.Validator = (*Config)(nil)
+
+// Validate validates the config.
+// This is automatically called by the config parser as it implements the xconfmap.Validator interface.
+func (cfg *Config) Validate() error {
+	if cfg.SamplesPerSecond < 1 {
+		return fmt.Errorf("invalid sampling frequency: %d", cfg.SamplesPerSecond)
+	}
+
+	if cfg.MapScaleFactor > MaxArgMapScaleFactor {
+		return fmt.Errorf(
+			"eBPF map scaling factor %d exceeds limit (max: %d)",
+			cfg.MapScaleFactor, MaxArgMapScaleFactor,
+		)
+	}
+
+	if cfg.BPFVerifierLogLevel > 2 {
+		return fmt.Errorf("invalid eBPF verifier log level: %d", cfg.BPFVerifierLogLevel)
+	}
+
+	if cfg.ProbabilisticInterval < 1*time.Minute || cfg.ProbabilisticInterval > 5*time.Minute {
+		return errors.New(
+			"invalid argument for probabilistic-interval: use " +
+				"a duration between 1 and 5 minutes",
+		)
+	}
+
+	if cfg.ProbabilisticThreshold < 1 ||
+		cfg.ProbabilisticThreshold > tracer.ProbabilisticThresholdMax {
+		return fmt.Errorf(
+			"invalid argument for probabilistic-threshold. Value "+
+				"should be between 1 and %d",
+			tracer.ProbabilisticThresholdMax,
+		)
+	}
+
+	if cfg.OffCPUThreshold < 0.0 || cfg.OffCPUThreshold > 1.0 {
+		return errors.New(
+			"invalid argument for off-cpu-threshold. The value " +
+				"should be in the range [0..1]. 0 disables off-cpu profiling")
+	}
+
+	if !cfg.NoKernelVersionCheck {
+		major, minor, patch, err := tracer.GetCurrentKernelVersion()
+		if err != nil {
+			return fmt.Errorf("failed to get kernel version: %v", err)
+		}
+
+		var minMajor, minMinor uint32
+		switch runtime.GOARCH {
+		case "amd64":
+			if cfg.VerboseMode {
+				minMajor, minMinor = 5, 2
+			} else {
+				minMajor, minMinor = 4, 19
+			}
```

Comment:
As stated in some other PR already - this is incorred.
`cfg.VerboseMode` does not have an effect on the minimum supported Linux kernel version.

Reply:
My previous comment: https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/859#discussion_r2414095558